pkgs/development/python-modules/antlr4-python3-runtime/default.nix

```
@@ -1,16 +1,24 @@
 { lib
 , buildPythonPackage
-, isPy3k
+, setuptools
 , python
-, antlr4 }:
+, antlr4
+}:
 
-buildPythonPackage rec {
+buildPythonPackage {
   pname = "antlr4-python3-runtime";
   inherit (antlr4.runtime.cpp) version src;
+
+  format = "pyproject";
+
   disabled = python.pythonOlder "3.6";
 
   sourceRoot = "source/runtime/Python3";
 
+  nativeBuildInputs = [
+    setuptools
+  ];
+
   # in 4.9, test was renamed to tests
```

I know this is not strictly related to your patch. Could you explain why is this `cd test*` not `cd tests`? Is this expression meant to be compatible with multiple antlr versions? If not, the `asterisk` could be removed.

---

I think the comment pretty much has explained it, so the folder is named `test` pre 4.9, but `tests` since 4.9.

---

> I think the comment pretty much has explained it, so the folder is named `test` pre 4.9, but `tests` since 4.9.

@NickCao what I tried to propose, is that perhaps because this expression is used with only a single `antlr4.runtime.cpp.src`, the asterisk could be removed.

---

But it can (and is) use with other versions of antlr, see `pkgs/servers/baserow/default.nix` for example.

---

> But it can (and is) used with other versions of antlr, see `pkgs/servers/baserow/default.nix` for example.

I see, than in that case, let's write this detail in the comment, something like:

```suggestion
  # We use an asterisk because this expression is used also for old antlr
  # versions, where there the tests directory is `tests` and not `test`. See
  # e.g in package `baserow`.
```